core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala

```diff
@@ -2,4 +2,18 @@ package com.databricks.labs.remorph.parsers.intermediate
 
 trait AstExtension
 
-case class Column(name: String) extends Expression with AstExtension {}
\ No newline at end of file
+case class Column(name: String) extends Expression with AstExtension {}
+
+abstract class Predicate extends Expression with AstExtension {}
+
+case class And(left: Predicate, right: Predicate) extends Predicate {}
+case class Or(left: Predicate, right: Predicate) extends Predicate {}
+case class Not(pred: Predicate) extends Predicate {}
```

Comment:
```suggestion
case class And(left: Expression, right: Expression) extends Binary(left, right) {}
case class Or(left: Expression, right: Expression) extends Binary(left, right) {}
case class Not(pred: Expression) extends Unary(pred) {}
```

because `1 > 2 AND TRUE` or `IS_MEMBER('admins') OR is_public`